core-text/src/font.rs

```diff
@@ -127,7 +129,21 @@ pub fn new_from_descriptor(desc: &CTFontDescriptor, pt_size: f64) -> CTFont {
 
 pub fn new_from_buffer(buffer: &[u8]) -> Result<CTFont, ()> {
     let ct_font_descriptor = create_font_descriptor(buffer)?;
-    Ok(new_from_descriptor(&ct_font_descriptor, 16.0))
+    Ok(new_from_name(&ct_font_descriptor.font_name(), 16.0)?)
```

Comment:
Won't this break if the font is not installed or if two different fonts use the same name? i.e. new_from_name won't find the font.

Reply:
Well, looks like WebRender also create CTFont from `postscript name`, see: https://phabricator.services.mozilla.com/D93518 
>        // there's no way great way to go from a CGFont to a CTFontDescriptor
>        // so we use the postscript name. Ideally NativeFontHandle would
>        // just use a CTFontDescriptor
>        let name = native_font_handle.0.postscript_name();
>        let font = core_text::font_descriptor::new_from_postscript_name(&name);
>
>        self.ct_font_descs
>            .insert(*font_key, font);

Also, [`font_name()` returns `postscript name`](https://developer.apple.com/documentation/coretext/kctfontnameattribute?language=objc), and [`new_fomr_name` receives `postscript name`](https://developer.apple.com/documentation/coretext/1509153-ctfontcreatewithname?language=objc).

Reply:
WebRender only uses the postscript name for installed system fonts. We don't use the name for data fonts:
https://searchfox.org/mozilla-central/source/gfx/wr/webrender/src/platform/macos/font.rs#382